File: webui/src/main/scala/com/taskadapter/webui/pages/AppUpdateNotificationComponent.scala
package com.taskadapter.webui.pages


import com.taskadapter.webui.service.CurrentVersionLoader
import com.taskadapter.webui.{LastVersionLoader, Page, VersionComparator}
import com.vaadin.flow.component.Component
import com.vaadin.flow.component.html.Label
import com.vaadin.flow.component.orderedlayout.HorizontalLayout



class AppUpdateNotificationComponent {
  val lastAvailableVersion = LastVersionLoader.loadLastVersion
  val currentVersion = new CurrentVersionLoader().getCurrentVersion

  def ui: Component = {
    val layout = new HorizontalLayout
    layout.setSpacing(true)
    layout.setHeight("50px")

    val outdated = VersionComparator.isCurrentVersionOutdated(currentVersion, lastAvailableVersion)
    if (outdated) {
      val message = Page.message("appUpdaterNotification.versionOutdated", currentVersion, lastAvailableVersion)
      val label = new Label(message)
      label.addClassName("important-notification-label")
      layout.add(label)
      val link = WebAppUpdater.createDownloadLink
      layout.add(link)
    }
    layout.setVisible(outdated)
    layout
  }
}

File: webui/src/main/java/com/taskadapter/webui/pages/AppUpdateNotificationComponent.java
package com.taskadapter.webui.pages;

import com.taskadapter.webui.LastVersionLoader;
import com.taskadapter.webui.Page;
import com.taskadapter.webui.VersionComparator;
import com.taskadapter.webui.service.CurrentVersionLoader;
import com.vaadin.flow.component.Component;
import com.vaadin.flow.component.html.Label;
import com.vaadin.flow.component.orderedlayout.VerticalLayout;

public class AppUpdateNotificationComponent extends VerticalLayout {

    public AppUpdateNotificationComponent() {

        String lastAvailableVersion = LastVersionLoader.loadLastVersion();
        String currentVersion = new CurrentVersionLoader().getCurrentVersion();

        boolean outdated = VersionComparator.isCurrentVersionOutdated(currentVersion, lastAvailableVersion);
        if (outdated) {
            String message = Page.message("appUpdaterNotification.versionOutdated", currentVersion, lastAvailableVersion);
            Label label = new Label(message);
            label.addClassName("important-notification-label");
            add(label);
            Component link = WebAppUpdater.createDownloadLink();
            add(link);
        }
        setVisible(outdated);
    }
}
